### PR TITLE
Snhut 584 user deactivate user account

### PR DESCRIPTION
**Pull request by joao-sd**
Description
===
This PR covers this [Jira Ticket](https://collectiveshift.atlassian.net/jira/software/c/projects/SNHUT/boards/132?modal=detail&selectedIssue=SNHUT-584&assignee=62901d109c617b006a6eec1c) and basically aims to implement an user activation/deactivation process.

Demo
===

https://user-images.githubusercontent.com/107008166/184690624-3f831cad-2f3a-481f-8a9e-424d89375c33.mp4

### CODE REVIEW

**Review by mk-sada — APPROVED** [review]
Separating user-table data-types was a nice touch!

PS/Note: This will require `rm -rf node_modules package-lock && npm install` to work properly.

File: apps/toboggan-app/src/app/shared/services/user/user.service.ts [hunk 1]
```diff
@@ -36,23 +31,8 @@ export class UserService {
...
-  enableUser() {
-    this.http.put('/api/users/:id/enable/', {}).subscribe(() => {
-      this.fetchUsers();
-    });
-  }
-
-  //
-  disableUser() {
-    this.http.put('/api/users/:id/disable', {}).subscribe(() => {
+  updateUser(updatedUser: IUpdatedUser, userId: string): void {
```
Comment (dmitriy-rodimin-sada) [inline apps/toboggan-app/src/app/shared/services/user/user.service.ts:34]: I think you would want to await the execution of this method in your component code. You can see how it is done above for the createNewUser. You can return a promise.

File: apps/toboggan-app/src/app/user/components/user-table/user-table.component.ts [hunk 2]
```diff
@@ -70,24 +81,105 @@ export class UserTableComponent {
...
+  ) {
+    this.userService.updateUser(
+      {
+        ...userPayload,
+        enabled: status === 'active',
+      },
+      userId
+    );
+
+    this.refreshTableData();
```
Comment (dmitriy-rodimin-sada) [inline apps/toboggan-app/src/app/user/components/user-table/user-table.component.ts:175]: if you don't await on 167, the 175 will run before user updates

**Review by dmitriy-rodimin-sada — CHANGES_REQUESTED** [review]

File: apps/toboggan-app/src/app/user/components/user-table/user-table.component.ts (hunk 2 above)
Comment (joao-sd) [reply apps/toboggan-app/src/app/user/components/user-table/user-table.component.ts:175]: Done

**Review by dmitriy-rodimin-sada — APPROVED** [review]

**Pull request merged**